core/Tracker/Visit.php

```diff
@@ -166,6 +166,12 @@ public function handle()
         // AND
         // - the last page view for this visitor was less than 30 minutes ago @see isLastActionInTheSameVisit()
         if (!$isNewVisit) {
+            // on a ping request that is received before the standard visit length, we just update the visit time w/o
+            // adding a new action
+            if ($this->isPingRequest()) {
+                $action = null;
+            }
+
```

Comment:
I'd really like to avoid modifying the tracker code a lot, since this method is already pretty complex. I think it would be easier to do this when https://github.com/piwik/piwik/issues/8071 is completed, then the modifications won't make the logic even more tangled.